src/os/bluestore/BlueFS.cc

```diff
@@ -3742,8 +4048,8 @@ int BlueFS::unlink(std::string_view dirname, std::string_view filename)
     return -EBUSY;
   }
   dir->file_map.erase(string{filename});
-  log_t.op_dir_unlink(dirname, filename);
-  _drop_link(file);
+  log.t.op_dir_unlink(dirname, filename);
+  _drop_link_D(file);
```

Comment:
Doesn't that produce N->D edge not covered in the lock graph? Is it effectively "unlink_LND"?

Generally we might want to tag (somehow differently, not in the func name. may be relevant comment in the caption?) all the public functions with that lock acquisition sequence as well  To ease missed sequences detecttion.

Reply:
> Generally we might want to tag (somehow differently, not in the func name, may be relevant comment in the caption?) all the public functions with that lock acquisition sequence as well To ease missed/wrong sequences detections.

May be the best approach would be having a properly "lock-seq-tagged" private counterpart for any public method which acquires the locks. And making public methods just a trivial single line wrappers.
E.g. for this func it would be:
int BlueFS::unlink() {
  return _unlink_LND();
}
...
_unlink_LND() {
....
}